File: 2018/day06/day06_test.go
package main

import (
	"testing"
)

func TestPart1(t *testing.T) {
	lines := readFile("day06.test.txt")

	expected := 17
	actual := part1(lines)
	if expected != actual {
		t.Fatalf("Expected %v but was %v", expected, actual)
	}
}

func TestPart2(t *testing.T) {
	lines := readFile("day06.test.txt")

	expected := 16
	actual := part2(lines, 32)
	if expected != actual {
		t.Fatalf("Expected %v but was %v", expected, actual)
	}
}

File: solutions/2025/day06/tests/day06_test.py
from day06.day06 import Day06
import unittest


class TestDay06(unittest.TestCase):
    def test_part1(self):
        input = Day06().read_file("day06/tests/day06.test.txt")
        self.assertEqual(Day06.solve_part1(input), 4277556)

    def test_part2(self):
        input = Day06().read_file("day06/tests/day06.test.txt")
        self.assertEqual(Day06.solve_part2(input), 3263827)


if __name__ == "__main__":
    unittest.main()
